src/main/scala/org/ergoplatform/nodeView/state/ErgoStateContext.scala

```diff
@@ -0,0 +1,58 @@
+package org.ergoplatform.nodeView.state
+
+import com.google.common.primitives.{Bytes, Shorts}
+import org.ergoplatform.modifiers.history.{Header, HeaderSerializer}
+import org.ergoplatform.settings.Constants
+import scorex.core.serialization.{BytesSerializable, Serializer}
+import scorex.crypto.authds.ADDigest
+
+import scala.util.Try
+
+/**
+  * Additional data, required for transactions validation.
+  *
+  * @param lastHeaders - fixed number of last applied header
+  * @param digest      - AVL tree root hash state BEFORE current block application
+  */
+case class ErgoStateContext(lastHeaders: Seq[Header], digest: ADDigest) extends BytesSerializable {
```

Comment:
I think this class and corresponding changes in the codebase should be moved into a separate PR, they have nothing to do with fixing todos.

Reply:
fixed